homeassistant/components/google/__init__.py

```diff
@@ -250,31 +249,30 @@ async def async_setup_services(
 ) -> None:
     """Set up the service listeners."""
 
-    created_calendars = set()
     calendars = await hass.async_add_executor_job(
         load_config, hass.config.path(YAML_DEVICES)
     )
 
-    async def _found_calendar(call: ServiceCall) -> None:
-        calendar = get_calendar_info(hass, call.data)
-        calendar_id = calendar[CONF_CAL_ID]
-
-        if calendar_id in created_calendars:
-            return
-        created_calendars.add(calendar_id)
-
+    async def _found_calendar(calendar_item: Calendar) -> None:
+        calendar = get_calendar_info(
+            hass,
+            {
+                **calendar_item.dict(exclude_unset=True),
+                CONF_TRACK: track_new,
+            },
+        )
         # Populate the yaml file with all discovered calendars
-        if calendar_id not in calendars:
-            calendars[calendar_id] = calendar
+        if calendar_item.id not in calendars:
+            calendars[calendar_item.id] = calendar
             await hass.async_add_executor_job(
                 update_config, hass.config.path(YAML_DEVICES), calendar
```

Comment:
I'm having trouble convincing myself this writing to the yaml file is safe given its starting multiple async tasks that schedule an executor appending to the same file.

Reply:
We could add a single `asyncio.Lock` for the YAML file.

Reply:
I'll send a followup to add this.

Reply:
Really, I want to come up with a plan to get off the google_calendars.yaml so maybe i'll focus on cleanup towards that goal instead.